lib/ClangImporter/ImportName.cpp

```diff
@@ -999,13 +999,13 @@ bool NameImporter::hasNamingConflict(const clang::NamedDecl *decl,
   lookupResult.setAllowHidden(true);
   lookupResult.suppressDiagnostics();
 
-  if (clangSema.LookupName(lookupResult, /*scope=*/nullptr)) {
+  if (clangSema.LookupName(lookupResult, /*scope=*/clangSema.TUScope)) {
     if (std::any_of(lookupResult.begin(), lookupResult.end(), conflicts))
       return true;
   }
 
   lookupResult.clear(clang::Sema::LookupTagName);
-  if (clangSema.LookupName(lookupResult, /*scope=*/nullptr)) {
+  if (clangSema.LookupName(lookupResult, /*scope=*/clangSema.TUScope)) {
```

Comment:
The second lookup performs a `LookupTagName`, which I believe should not produce more lookup results in C++ (In C it can though). In that case , do we really need to adjust the scope here? Also, if this lookup isn't needed in C++, can we just wrap it in an `if` when clang's lang options have C++ on to remove the redundant lookup?

Reply:
I can try both theories and I see if it changes any result.

I found the problem with the first, and I thought that keeping both with the same scope made sense, but it might not be so.

Reply:
Neither changing back to `nullptr`, nor wrapping the lookup for a check of C++ changed the result of the test, or the results in the rest of the test suite, so it seems safe to assume that `LookupTagName` is not necessary. Thanks for the help!